src/Dompdf.php

```diff
@@ -409,31 +410,38 @@ public function loadHtmlFile($file)
      * @param string $encoding
      * @deprecated
      */
-    public function load_html($str, $encoding = 'UTF-8')
+    public function load_html($str, $encoding = null)
     {
         $this->loadHtml($str, $encoding);
     }
 
     /**
      * Loads an HTML string
      * Parse errors are stored in the global array _dompdf_warnings.
-     * @todo use the $encoding variable
      *
      * @param string $str HTML text to load
-     * @param string $encoding Not used yet
+     * @param string $encoding Encoding of $str
      */
-    public function loadHtml($str, $encoding = 'UTF-8')
+    public function loadHtml($str, $encoding = null)
     {
         $this->saveLocale();
 
-        // FIXME: Determine character encoding, switch to UTF8, update meta tag. Need better http/file stream encoding detection, currently relies on text or meta tag.
-        $known_encodings = mb_list_encodings();
-        mb_detect_order('auto');
-        if (($file_encoding = mb_detect_encoding($str, null, true)) === false) {
-            $file_encoding = "auto";
+        // FIXME: Determine character encoding, switch to UTF8, update meta tag. Need better http/file stream encoding detection when $encoding parameter not used, currently relies on text or meta tag.
```

Comment:
Fixme parts of this comment should be removed.

Reply:
Detect enconding when not specified still need to be improved. Do you have a better comment to this part? I couldn't think in a better comment.

Reply:
I improved this comment, but still accepting suggestions!

Reply:
Form my point of view the Fixme comment mainly describes the not working content type detection from loadHtmlFile which calls this loadHtml method. This is something that was indeed broken and you solved here.
An even better detection is just on top but not worth a fixme comment. What could the better detection on html document level be in your opinion? Actually using the meta information instead of just rewriting it? I wondered about this current functionally at least.

Reply:
I understood your point. Just a moment and i'll fix this comment.